scene/2d/tile_map.cpp

```diff
@@ -1269,7 +1269,8 @@ void TileMap::_set_tile_data(const PoolVector<int> &p_data) {
 	for (int i = 0; i < c; i += offset) {
 		const uint8_t *ptr = (const uint8_t *)&r[i];
 		uint8_t local[12];
-		for (int j = 0; j < ((format == FORMAT_2) ? 12 : 8); j++) {
+		const int buffer_size = (format == FORMAT_2) ? 12 : 8;
+		for (int j = 0; j < buffer_size; j++) {
```

Comment:
This was needed to workaround a GCC false positive:
```
 scene/2d/tile_map.cpp: In member function 'void TileMap::_set_tile_data(const PoolVector<int>&)':
Error: scene/2d/tile_map.cpp:1273:34: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1273 |                         local[j] = ptr[j];
      |                         ~~~~~~~~~^~~~~~~~
scene/2d/tile_map.cpp:1271:25: note: at offset 12 into destination object 'local' of size 12
 1271 |                 uint8_t local[12];
      |                         ^~~~~
```

https://gcc.gnu.org/bugzilla/show_bug.cgi?id=104789